src/main/java/org/jabref/JabRefMain.java

```diff
@@ -119,9 +118,10 @@ private static void ensureCorrectJavaVersion() {
                 versionError.append("\n");
                 versionError.append(Localization.lang("Note that currently, JabRef does not run with Java 9."));
             }
-            final JFrame frame = new JFrame();
-            JOptionPane.showMessageDialog(null, versionError, Localization.lang("Error"), JOptionPane.ERROR_MESSAGE);
-            frame.dispose();
+
```

Comment:
I think this dialog is superflous as JabRef won't even start with java9 cause of the class/module visibility